### ISSUE TITLE
Service always dies immidiately

### ISSUE THREAD

**Issue by Delegator9176** [YunoHost-Apps/mautrix_signal_ynh#84]
### Describe the bug
After the last update to version 0.4.3 mautrix-signal stopped working the service is always crashing. The log shows there is an issue with the signal-bot system user and his keys

here are the logs from the mautrix-signal-service
https://paste.yunohost.org/yigapolote

Thanks in advance

**Comment by TiberiusRegenschein** [comment]
I can confirm this behavior

**Comment by CodeShakingSheep** [comment]
I have the same behavior. See my logs here: https://paste.yunohost.org/anohexibor

**Comment by Delegator9176** [comment]
> I can confirm this behavior. A solution can be found here: https://forum.yunohost.org/t/mautrix-signal-hs-suite-a-mise-a-jour/25991/5

1. Edit the file:  `nano /etc/yunohost/apps/mautrix_signal/settings.yml `
2. retrieve the value of psqlpwd then copy this value 
3. Goto the file: `nano /opt/yunohost/mautrix_signal/config.yaml`
4. Find the line containing `database: postgres://mautrix_signal@localhost:5432/mautrix_signal`
5. and modify it so that it contains your password in front of the @ with a leading `:` -> `database: postgres://mautrix_signal:TRUCSECRETCOPIERAVANT@localhost:5432/mautrix_signal`
6. Restart service and all is OK: `systemctl restart mautrix_signal`

**Comment by Gredin67** [comment]
Is this really solved by any PR ? Or you just proposed a workaround ? Did you have a look @nathanael-h ?

**Comment by nathanael-h** [comment]
Hello, I did not had a look yet. Thanks for the ping. I don't really understand why we have this issue here on mautrix_signal_ynh which is the same as we had on mautrix_whatsapp_ynh, as the latestet upgrade of the app just changed the sources and the `config.yaml` files

I see in the upgrade script that the `db_pwd` variable is initialised two times: 

https://github.com/YunoHost-Apps/mautrix_signal_ynh/blob/139a88eb8986b8b15e5bd9f120ec615991966e6b/scripts/upgrade#L28

https://github.com/YunoHost-Apps/mautrix_signal_ynh/blob/139a88eb8986b8b15e5bd9f120ec615991966e6b/scripts/upgrade#L74

It seems that at the time of chaging the key to store the db password value, from `db_pwd` to `psqlpwd` we forgot to handle this switch in the upgrade process. What do you think about this? I could send a PR.

**Comment by Gredin67** [comment]
yep, obviously nobody sorted this out https://github.com/YunoHost-Apps/mautrix_signal_ynh/issues/67
https://github.com/YunoHost-Apps/mautrix_whatsapp_ynh/issues/84
https://github.com/YunoHost-Apps/mautrix_telegram_ynh/pull/43/files
https://github.com/YunoHost-Apps/mautrix_whatsapp_ynh/pull/85

**Comment by MayeulC** [comment]
I agree it's annoying, I dug into it some time ago, and didn't really understand how the configuration was handled. There seems to be a lot of "dark magic" stuff that gets automatically handled by yunohost implicitly, which makes understanding everything complicated.

A big refactor is needed, possibly with packaging v2. I would also like to keep migration operations grouped together in the code. Unfortunately, the changes were not really documented, so we'd have to go over the code history and write down configuration changes. There is definitely a need to clean up unused variables as per #67, and better handle version bumps.

It doesn't help that upstream is being rewritten, and as a result the packaging will have to be completely overhauled at some point.

Finally, I am not sure why I keep missing github notifications. Probably because I have so many of these? I apologize for missing this issue.
